portfolio/src/main/webapp/script.js

```diff
@@ -37,22 +37,154 @@ function addRandomGreeting() {
 }
 
 function getPageComments() {
-  fetch('/data').then(response => response.json()).then((list) => {
-    const listElement = document.getElementById('comments-container');
-    listElement.innerHTML = '';
-
-    for(i = 0; i < list.length; i++) {
-        if(list[i] == "") {
-            continue;
-        } else {
-            listElement.innerText = list[i];
+    const text = document.getElementById('text').value;
+    const languageCode = document.getElementById('language').value;
+
+    const resultContainer = document.getElementById('comments-container');
+    resultContainer.innerText = 'Loading...';
+
+    const params = new URLSearchParams();
+    params.append('text', text);
+    params.append('languageCode', languageCode);
+
+    fetch('/data').then(response => response.json()).then((list) => {
+        const listElement = document.getElementById('comments-container');
+        listElement.innerHTML = '';
+
+        for(i = 0; i < list.length; i++) {
+            if(list[i] == "") {
+                continue;
```

Comment:
An improvement you can make here, is instead of filtering out the comments that are empty, validate that comments aren't empty before putting them in the database :) so when anyone adds a comment, your POST method could verify that the string isn't empty - and if it is, don't put it in the database at all. That way you're catching and validating the comment on the POST side rather than on the GET side, which is earlier, and prevents you from having empty comments in your database.